pandas/tests/frame/test_constructors.py

```diff
@@ -2296,6 +2296,70 @@ def test_constructor_list_str_na(self, string_dtype):
         expected = DataFrame({"A": ["1.0", "2.0", None]}, dtype=object)
         tm.assert_frame_equal(result, expected)
 
+    @pytest.mark.parametrize("copy", [False, True])
+    @td.skip_array_manager_not_yet_implemented
+    def test_dict_nocopy(self, copy, any_nullable_numeric_dtype, any_numpy_dtype):
+        a = np.array([1, 2], dtype=any_numpy_dtype)
+        b = np.array([3, 4], dtype=any_numpy_dtype)
+        if b.dtype.kind in ["S", "U"]:
+            # These get cast, making the checks below more cumbersome
+            return
+
+        c = pd.array([1, 2], dtype=any_nullable_numeric_dtype)
+        df = DataFrame({"a": a, "b": b, "c": c}, copy=copy)
+
+        def check_views():
+            # written to work for either BlockManager or ArrayManager
+            assert sum(x is c for x in df._mgr.arrays) == 1
+            assert (
+                sum(
+                    x.base is a for x in df._mgr.arrays if isinstance(x.dtype, np.dtype)
+                )
+                == 1
+            )
+            assert (
+                sum(
+                    x.base is b for x in df._mgr.arrays if isinstance(x.dtype, np.dtype)
+                )
+                == 1
+            )
+
+        if not copy:
+            # constructor preserves views
+            check_views()
+
+        df.iloc[0, 0] = 0
+        df.iloc[0, 1] = 0
+        if not copy:
+            # setitem on non-EA values preserves views
+            assert sum(x is c for x in df._mgr.arrays) == 1
```

Comment:
I don't really understand that comment, which is about non_EA, but the assert itself is only testing that the EA is still a view?

Reply:
Can you see if you can clarify this code/comment a bit?

Reply:
edited.  LMK if still unclear

Reply:
Thanks, yes, that clarifies